api/migrations/20200519165645-ocersData.js

```diff
@@ -0,0 +1,996 @@
+    sourceDateUpdated: new Date(),
+    sourceSystemRef: 'ocers-csv'
+  };
+
+  const responseMaster = await nrptiCollection.insertOne(masterRecord);
+};
+
+const getIssuedToObject = function (row, addPublicRole) {
+  let issuedToObject = {
+    read: ['sysadmin'],
+    write: ['sysadmin']
+  };
+
+  if (!row) {
+    return issuedToObject;
+  }
+
+  if (row[2]) {
+    issuedToObject.type = 'Company';
+    issuedToObject.companyName = row[2];
+
+    issuedToObject.fullName = setIssuedToFullNameValue(issuedToObject);
+
+    if (addPublicRole) {
+      issuedToObject.read.push('public');
+    }
+  } else if (row[3] || row[4] || row[5]) {
+    issuedToObject.type = 'Individual';
+    issuedToObject.firstName = row[3];
+    issuedToObject.middleName = row[5];
+    issuedToObject.lastName = row[4];
+    issuedToObject.dateOfBirth = null;
+
+    issuedToObject.fullName = setIssuedToFullNameValue(issuedToObject);
+
+    if (addPublicRole) {
+      issuedToObject.read.push('public');
+    }
+  } else {
+    // Individual is anonymous, so don't set read role
+    issuedToObject.type = 'Individual';
+    issuedToObject.firstName = '';
+    issuedToObject.middleName = '';
+    issuedToObject.lastName = '';
+    issuedToObject.dateOfBirth = null;
+
+    issuedToObject.fullName = '';
+  }
+
+  return issuedToObject;
+};
+
+const setIssuedToFullNameValue = function (issuedToObj) {
+  if (!issuedToObj || !issuedToObj.type) {
+    return '';
+  }
+
+  if (issuedToObj.type === 'IndividualCombined') {
+    return issuedToObj.fullName;
+  }
+
+  if (issuedToObj.type === 'Company') {
+    return issuedToObj.companyName;
+  }
+
+  if (!issuedToObj.firstName && !issuedToObj.middleName && !issuedToObj.lastName) {
+    return '';
+  }
+
+  if (issuedToObj.type === 'Individual') {
+    let entityString = '';
+
+    const entityNameParts = [];
+    if (issuedToObj.lastName) {
+      entityNameParts.push(issuedToObj.lastName);
+    }
+
+    if (issuedToObj.firstName) {
+      entityNameParts.push(issuedToObj.firstName);
+    }
+
+    entityString = entityNameParts.join(', ');
+
+    if (issuedToObj.middleName) {
+      entityString += ` ${issuedToObj.middleName}`;
+    }
+
+    return entityString;
+  }
+};
+
+const getPenaltyObject = function (row) {
+  const penaltyObject = {
+    type: '',
+    penalty: {
+      type: '',
+      value: null
+    },
+    description: ''
+  };
+
+  if (!row) {
+    return penaltyObject;
+  }
+
+  if (row[[14]]) {
+    penaltyObject.type = 'Fined';
+    penaltyObject.penalty.type = 'Dollars';
+    penaltyObject.penalty.value = Number(row[14]);
+  } else {
+    penaltyObject.type = 'Other';
+    penaltyObject.penalty.type = 'Other';
+  }
+
+  penaltyObject.description = row[15] || '';
+
+  return [penaltyObject];
+};
+
+const calculateDateFormat = function (row) {
+  let dateIssued = null;
+  if (row[1]) {
+    if (moment(row[1], 'MM/DD/YYYY', true).isValid()) {
+      dateIssued = moment(row[1], 'DD/MM/YYYY').toDate();
```

Comment:
Why are we checking MM/DD/YYYY but then creating DD/MM/YYYY?

Reply:
This is the date format issue I commented about in the description of the PR. 

It was an issue in the original nrced csv migration, which I fixed in the new ocers migration, in the second commit.